### ISSUE TITLE
Scaffold body missing when coming from JsonWidgetData.recreate()

### ISSUE THREAD

**Issue by dan461** [peiffer-innovations/json_dynamic_widget#129]
Hello. First off I want to say thanks for the time and effort put into this package. It's impressive work and it may open up some impactful options for the project I'm working on.

I'm working on a proof-of-concept Flutter app and in this app I have two screens, I'll call them Master and Detail, both created from json using the package. They both have Scaffolds and I push from Master to Detail and the nav behavior is typical. 
The json for the screens comes from a single service call and is stored in a repo. When pushing to Detail the 'detail_json' is provided from the repo to create the screen. This all works fine. The json is structured in general like this:

```
{
    "widget_json": {
        "master_json": {
            "type": "scaffold",
            "args": {
                "appBar": {...},
                "body": {...}
            }
        },
        "detail_json": {
            "type": "scaffold",
            "args": {
                "appBar": {...},
                "body": {...}
            }
        }
    }
}
```

Here is the issue I've found.
When you first go from Master to Detail all is well. The scaffold on Detail has its body. If you return to Master (back button) and then go back to the Detail page the Scaffold will have no body. (there's no service call done here, since the json for Detail is in the repo)

Here is what I discovered:
In JsonWidgetData.fromDynamic() a Scaffold will have its 'body' value used as its child, and the body is then removed from the args map. When you return to the Detail page the Scaffold is recreated by the JsonWidgetData.recreate() but it has no originalChild so the Scaffold will have no body. 
This seems to be due to it not having its originalChild set in fromDynamic, since there is no map['child'] to use. I found that setting the map['child'] to the scaffold's body before removing the body from the map fixes the issue. Not removing the args['body'] also fixes it. I haven't looked much further into this so don't know if either fix would cause other issues, so I wanted to share what I found. Let me know what you think. Thanks!

```
if (type == 'scaffold' && map['args'] is Map && child == null) {
          child = JsonWidgetData.fromDynamic(
            map['args']['body'],
            registry: registry,
          );

          var args = Map<String, dynamic>.from(map['args']);

         // POSSIBLE FIX
          map['child'] = args['body'];
          args.remove('body');
          map['args'] = args;
        }
```

**Comment by crisperit** [comment]
Hey - really thanks for in depth view 

The Scaffold implementation is tricky - I'll create a PR with your fix and we'll debate with @jpeiffer if that will work for all cases

**Comment by jpeiffer** [comment]
Merged and deployed as 6.0.1.

Closing ticket, please reopen or create a new one if it is still causing you issues.
